snapcraft/internal/lxd.py

```diff
@@ -147,3 +134,47 @@ def _wait_for_network(self):
                 if retry_count == 0:
                     raise e
         logger.info('Network connection established')
+
+
+def _get_default_remote():
+    """Query and return the default lxd remote.
+
+    Use the lxc command to query for the default lxd remote. In most
+    cases this will return the local remote.
+
+    :returns: default lxd remote.
+    :rtype: string.
+    :raises snapcraft.internal.errors.SnapcraftEnvironmentError:
+        raised if the lxc call fails.
+    """
+    try:
+        default_remote = check_output(['lxc', 'remote', 'get-default'])
```

Comment:
I like this because it's python's ask for forgiveness, not for permission.
But there might be other reasons here. Like lxd is installed, but you are not in the lxd group. You are swallowing the error that lxc returns, so there is no way to know it.
Maybe put a more generic message, and print what lxc returns as an error?